### DISCUSSION TITLE
Is there a way to use the same random voice multiple times?

### DISCUSSION THREAD

**Discussion by Talvara** [neonbjb/tortoise-tts#455] [Q&A]
So I'm not super interested in voice cloning, but I am interested in generating voice lines for Indy game development.

What I would like to achieve is to find a good --voice Random and then reuse that voice to create a potentially infinite amount of voice lines that sound like they are spoken by the same character.

What I've tried thus far is adding a --seed and limiting the --candidates to 1:
`python tortoise/do_tts.py --text "This is a random voice with the seed: 1234, cool right?" --candidates 1 --voice random --preset fast --seed 1234`

And this will generate the text line with the same voice every time, but when a different text line is provided, the random voice changes with it.

I tried digging into the .py scripts to see if I could find where the random voice vector is created.
Which I think is line396 of the API.py script.
I figured maybe the amount of times torch.randn was different by the time the execution came to this line so I tried setting the manual seed for torch just before this line like so:
`        auto_conds = None
        if voice_samples is not None:
            auto_conditioning, diffusion_conditioning, auto_conds, _ = self.get_conditioning_latents(voice_samples, return_mels=True)
        elif conditioning_latents is not None:
            auto_conditioning, diffusion_conditioning = conditioning_latents
        else:
            **deterministic_seed = self.deterministic_state(seed=use_deterministic_seed)**
            auto_conditioning, diffusion_conditioning = self.get_random_conditioning_latents()`
            
But this hasn't made a difference.

Is there a way to reuse the random voice from one line for more lines of text?

Kind regards,
~Talvara

**Comment by ooofbrain** [comment]
you change the --voice random to --voice voicename for the same voice. for the lines of speech your character speaks you need a lm something like a 6b or 7b parameter wizardml lm might be handy
